scheme/arm-cca/store_handler.go

```diff
@@ -0,0 +1,85 @@
+// Copyright 2024 Contributors to the Veraison project.
+// SPDX-License-Identifier: Apache-2.0
+
+package arm_cca
+
+import (
+	"fmt"
+
+	"github.com/veraison/ccatoken"
+	"github.com/veraison/services/handler"
+	"github.com/veraison/services/proto"
+	"github.com/veraison/services/scheme/common/arm"
+	realm "github.com/veraison/services/scheme/common/cca/realm"
+)
+
+type StoreHandler struct{}
+
+func (s StoreHandler) GetName() string {
+	return "cca-store-handler"
+}
+
+func (s StoreHandler) GetAttestationScheme() string {
+	return SchemeName
+}
+
+func (s StoreHandler) GetSupportedMediaTypes() []string {
+	return nil
+}
+
+func (s StoreHandler) SynthKeysFromRefValue(
+	tenantID string,
+	refVal *handler.Endorsement,
+) ([]string, error) {
+	switch refVal.SubType {
+	case "sw-component", "platform-config":
+		return arm.SynthKeysForPlatform(SchemeName, tenantID, refVal)
+	case "realm.reference-value":
+		return realm.SynthKeysForCcaRealm(SchemeName, tenantID, refVal)
+	default:
+		return nil, fmt.Errorf("invalid SubType: %s, for Scheme: %s", refVal.SubType, refVal.Scheme)
+	}
+}
+
+func (s StoreHandler) SynthKeysFromTrustAnchor(tenantID string, ta *handler.Endorsement) ([]string, error) {
+	return arm.SynthKeysFromTrustAnchors(SchemeName, tenantID, ta)
+}
+
+func (s StoreHandler) GetTrustAnchorIDs(token *proto.AttestationToken) ([]string, error) {
+	var evidence ccatoken.Evidence
+	err := evidence.FromCBOR(token.Data)
+	if err != nil {
+		return []string{""}, handler.BadEvidence(err)
+	}
+
+	claims := evidence.PlatformClaims
+	if err != nil {
+		return []string{""}, err
+	}
+	taID, err := arm.GetTrustAnchorID(SchemeName, token.TenantId, claims)
+	if err != nil {
+		return []string{""}, err
+	}
+
+	return []string{taID}, nil
+}
+
+func (s StoreHandler) GetRefValueIDs(
+	tenantID string,
+	trustAnchors []string,
+	claims map[string]interface{},
+) ([]string, error) {
+	platformClaimsMap, ok := claims["platform"].(map[string]interface{})
+	if !ok {
+		return nil, fmt.Errorf("claims do not contain platform map: %v", claims)
+	}
+	pids, err := arm.GetPlatformReferenceIDs(SchemeName, tenantID, platformClaimsMap)
+	if err != nil {
+		return nil, fmt.Errorf("unable to get cca platform reference IDs: %w", err)
+	}
+	rids, err := realm.GetRealmReferenceIDs(SchemeName, tenantID, claims)
```

Comment:
Why is the realm looking at the whole claims-set rather than just the realm part?

Reply:
Yes, this should mirror how platform part is implemented, I shall change this, to be more clear!

Reply:
Corrected in: https://github.com/veraison/services/commit/33f2ed4b6e6aeb7a20a3582a00a9445d97276291